tests/bud.bats

```diff
@@ -1796,3 +1796,29 @@ load helpers
   run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t ${target} git://github.com/containers/BuildSourceImage#master
   run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t ${target} git://github.com/containers/BuildSourceImage
 }
+
+@test "bud containerfile with tar archive in copy" {
+  # First check to verify cache is uses if the tar file does not change
+  target=copy-archive
+  date>bud/${target}/test; tar cJf - bud/${target}/test > bud/${target}/test.tar.xz
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t ${target} ${TESTSDIR}/bud/${target}
+  expect_output --substring "COMMIT copy-archive"
+
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t ${target} ${TESTSDIR}/bud/${target}
+  expect_output --substring " Using cache"
+  expect_output --substring "COMMIT copy-archive"
+
+  # Now test that we do NOT use cache if the tar file changes
+  date>bud/${target}/test; tar cJf - bud/${target}/test > bud/${target}/test.tar.xz
```

Comment:
Another holdover comment from #1929: I kind of cringe at using `date >` in this situation: it is not impossible for lines 1803-1812 to run in less than one second, in which case the `test` file mtime might change but its content would not. I would like to see a clearer, more explicit change: maybe something like
```
  echo This is new content >> bud/${target}/test; tar ....
```